src/Components/Input/Input.tsx

```diff
@@ -47,47 +47,50 @@ interface Props extends TextInputProps {
     | "decimal-pad"
     | "twitter"
     | "web-search";
-  inputFontSize: number;
   errorFontSize: 12;
   errorColor: string;
-  borderBottomWidth: number;
+  style?: IFieldTextStyle;
+  textInputProps?: TextInputProps;
 }
 
 type State = {
   labelFontSize: Animated.Value;
   labelMarginBottom: Animated.Value;
+  labelMarginTop: Animated.Value;
   value?: string;
   error: string | void;
 };
 
+const DEFAULT_INPUT_FONT_SIZE = 14;
+const DEFAULT_INPUT_PADDING_VERTICAL = 2;
+
 class Input extends React.PureComponent<Props, State> {
-  textInput?: TextInput;
+  private textInput?: TextInput;
+  private animationLabelUp;
+  private animationLabelDown;
+
   static defaultProps = {
     onSuccessInputFieldColor: Colors.themeColor,
     textColor: Colors.defaultText,
     secureTextEntry: false,
     keyboardType: "default",
     width: "100%",
-    height: 84,
-    maxLabelFontSize: 14,
-    minLabelFontSize: 12,
-    maxLabelMarginBottom: isWeb ? 16 : 0,
-    minLabelMarginBottom: isWeb ? -24 : isAndroid ? -34 : -28,
-    inputFontSize: 14,
     errorFontSize: 12,
     errorColor: Colors.error,
-    borderBottomWidth: 1,
   };
 
+  private inputFontSize = this.getInputFontSize();
+  private inputPaddingVertical = this.getInputPaddingVertical();
+  private labelInitialMargin = -(this.inputFontSize + this.inputPaddingVertical);
```

Comment:
There should be a better way to style the component without the need of calculating the margin from those styles.

And without forcing the consumer to pass paddingVertical in fixed number only